discovery_engine_core/core/src/engine.rs

```diff
@@ -68,7 +68,7 @@ struct Market {
 pub struct Config {
     api_key: String,
     api_base_url: String,
-    markets: Vec<Market>,
+    markets: RwLock<Vec<Market>>,
```

Comment:
We cannot do this because it will complicate how the ffi can work with this struct. 
We can keep the relevant data in another struct and use this only as a way of passing messages between dart and rust.
In this new struct we can drop the paths to the models.